addons/account/models/account_move.py

```diff
@@ -4450,7 +4457,10 @@ def fix_remaining_cent(currency, abs_residual, partial_amount):
 
             # Move to the next available credit line.
             if not credit_line:
-                credit_line = next(credit_lines, None)
+                if not credit_records:
+                    credit_line = next(void_lines, None)
```

Comment:
Are you sure this works? calling `next` *consumes* the iterator, so if a record is called above in  `debit_line = next(void_lines, None)`, it will not be available in here.

Reply:
Yes, I tried to do the same as in https://github.com/odoo/odoo/pull/92666/files, in a sense. Usually, in case void lines exist, they come in pairs. Thus, the void line fetched as debit_line and the void  line fetched as credit_line will be reconciled.

Reply:
I think "*usually* come in pairs" is not assuring.

Reply:
I'm not sure I get the point?
Isn't it exactly the point of creating an iterator, to consume it and not get the same line twice?

Reply:
Credit lines and debit lines are completely distinguishable, but the void lines, you just put them all in a one iterator and fetch them in two different places, I don't know what the contents is, I am just saying the pattern looks suspicious.

Reply:
It will be reconciled with the debit or credit, depending on in which group it is assigned.
It will always be reconciled once and only once.

Reply:
Yep, I thought it worth a double check ;)